src/test/java/core/basesyntax/service/OperationStrategyImplTest.java

```diff
@@ -0,0 +1,76 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.operation.BalanceOperation;
+import core.basesyntax.service.operation.OperationHandler;
+import core.basesyntax.service.operation.PurchaseOperation;
+import core.basesyntax.service.operation.ReturnOperation;
+import core.basesyntax.service.operation.SupplyOperation;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class OperationStrategyImplTest {
+
+    private OperationStrategyImpl operationStrategy;
+
+    @BeforeEach
+    void setUp() {
+
+        OperationHandler balanceHandler = new BalanceOperation();
+        OperationHandler supplyHandler = new SupplyOperation();
+        OperationHandler returnHandler = new ReturnOperation();
+        OperationHandler purchaseHandler = new PurchaseOperation();
+
+        Map<FruitTransaction.Operation, OperationHandler> operationHandlerMap = new HashMap<>();
+        operationHandlerMap.put(FruitTransaction.Operation.BALANCE, balanceHandler);
+        operationHandlerMap.put(FruitTransaction.Operation.SUPPLY, supplyHandler);
+        operationHandlerMap.put(FruitTransaction.Operation.PURCHASE, purchaseHandler);
+        operationHandlerMap.put(FruitTransaction.Operation.RETURN, returnHandler);
+
+        operationStrategy = new OperationStrategyImpl(operationHandlerMap);
+    }
+
+    @Test
+    void shouldReturnBalanceHandler_whenOperationIsBalance() {
+        OperationHandler handler = operationStrategy.get(FruitTransaction.Operation.BALANCE);
+        assertNotNull(handler, "Handler for BALANCE should not be null");
+        assertTrue(handler instanceof BalanceOperation,
```

Comment:
What is the difference between getClass() and instanceof? What will be better to use in this case?

Reply:
better to use instanceof, since it allows more flexibility in dealing with possible class descendants.